### PR TITLE
168 feature request add formula 516 from nen en 1992 1 1 c2 2011

### PR DESCRIPTION
**Pull request by MaartenBW**
## Description

Please write a description of the changes you made here.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x ] I have added tests that prove my fix is effective or that my feature works
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [ x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: blueprints/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/formula_5_14.py [hunk 1]
```diff
@@ -26,6 +26,8 @@ def __init__(
         ----------
         l_0 : M
             [:math:`l_{0}`] Effective length [:math:`m`].
+            Use your own implementation of this value or use :class: `Form5Dot15EffectiveLengthBraced`
+            or :class: `Form5Dot15EffectiveLengthUnbraced`.
```
Comment (bro-wi) [inline blueprints/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/formula_5_14.py:30]: I think we have to add an 'integration' test to support this always.

File: blueprints/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/formula_5_15.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        NEN-EN 1992-1-1+C2:2011 art.5.8.3.2(1) - Formula (5.15)
+
+        Parameters
+        ----------
+        k_1 : -
+            [:math:`k_{1}`] Relative flexibility of rotational constraint at end 1 [:math:`-`].
+        k_2 : -
+            [:math:`k_{1}`] Relative flexibility of rotational constraint at end 2 [:math:`-`].
+        l : M
```
Comment (bro-wi) [inline blueprints/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/formula_5_15.py:29]:
```suggestion
        height : M
```

File: blueprints/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/formula_5_15.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+from blueprints.validations import raise_if_negative
+
+
+class Form5Dot15EffectiveLengthBraced(Formula):
+    """Class representing formula 5.15 for the calculation of the slenderness ratio, :math:`λ`."""
+
+    label = "5.15"
+    source_document = NEN_EN_1992_1_1_C2_2011
+
+    def __init__(self, k_1: DIMENSIONLESS, k_2: DIMENSIONLESS, height: M) -> None:
```
Comment (bro-wi) [inline blueprints/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/formula_5_15.py:18]: I think it is a good choice to change l (lower L) into 'height'!

**Review by bro-wi — APPROVED** [review]
Nice implementation. Only some small comments on my part.

File: blueprints/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/formula_5_15.py (hunk 2 above)
Comment (MaartenBW) [reply blueprints/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/formula_5_15.py:29]: Fixed in PR #272

File: tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/test_formula_5_15.py
Comment (egarciamendez) [inline tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/test_formula_5_15.py:1]: Not the same as the eurocode 
afbeelding

afbeelding

File: tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/test_formula_5_16.py
Comment (egarciamendez) [inline tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/test_formula_5_16.py:1]: Same goes for this one.
afbeelding
afbeelding

**Review by egarciamendez — CHANGES_REQUESTED** [review]
Found a lot of errors in the docs and latex representation. Need to fix before merging, I'm fixing those already but it's getting kind of late. Will continue tomorrow.

**Review by egarciamendez — APPROVED** [review]
ready after some fixes

**Pull request merged**